[ML] Add index pattern management to index Data Visualizer

qn895 opened the pull request:
## Summary

Part of https://github.com/elastic/kibana/issues/101435 and follow up of https://github.com/elastic/kibana/pull/100922. This PR adds the index pattern field editor flyout to the Data Visualizer. New functionalities include:

<img width="758" alt="Screen Shot 2021-06-04 at 16 03 42" src="https://user-images.githubusercontent.com/43350163/120952261-4b337600-c710-11eb-8c43-38a3f56bcb58.png">

- [x] Add new index pattern field
![2021-06-06 at 22 40](https://user-images.githubusercontent.com/43350163/120956047-60140780-c718-11eb-9c25-8e60eaa8d63b.gif)

- [x] Edit index pattern field
![2021-06-06 at 22 34](https://user-images.githubusercontent.com/43350163/120955625-69e93b00-c717-11eb-904c-1f2c77a4c458.gif)

- [x] Manage index pattern fields
![2021-06-06 at 22 36](https://user-images.githubusercontent.com/43350163/120955740-afa60380-c717-11eb-8d42-3f82fe6cd56b.gif)

- [x] Delete index pattern field

- [x] Added tooltip for the Show/Hide distribution button

<img width="271" alt="Screen Shot 2021-06-16 at 17 20 41" src="https://user-images.githubusercontent.com/43350163/122302115-48cadc00-cec7-11eb-9380-38eda4723ce6.png">

### Review Hints:
- **kibana-app-services**: [Added `data-test-subj` for index pattern editor form](src/plugins/index_pattern_field_editor/public/components/field_editor/field_editor.tsx) to help with functional tests

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/)
* Commit: 76b7dffd497b14c40e603af1931c03370beb12ee
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Bundle Limits](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/execution/node/410/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/)
* Commit: ca85c18f2bff94bc69f59b63cb2a42049e3d5e55
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Plugin List Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/execution/node/427/log)

### History
* :broken_heart: [Build #129003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/) failed 76b7dffd497b14c40e603af1931c03370beb12ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/)
* Commit: 5e4c520b371288d46408e5909fc1eb9a647e3348
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Plugin List Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/execution/node/420/log)

### History
* :broken_heart: [Build #129060](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/) failed ca85c18f2bff94bc69f59b63cb2a42049e3d5e55
* :broken_heart: [Build #129003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/) failed 76b7dffd497b14c40e603af1931c03370beb12ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/)
* Commit: fbbf1e923ffedd3532419450f299f581b97d31a9
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/execution/node/423/log)

### History
* :broken_heart: [Build #129297](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/) failed 5e4c520b371288d46408e5909fc1eb9a647e3348
* :broken_heart: [Build #129060](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/) failed ca85c18f2bff94bc69f59b63cb2a42049e3d5e55
* :broken_heart: [Build #129003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/) failed 76b7dffd497b14c40e603af1931c03370beb12ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/)
* Commit: e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f466ebf1a3fc33ea09ea51853811fe3a8f77c1d2) | [after](https://github.com/elastic/kibana/commit/e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f466ebf1a3fc33ea09ea51853811fe3a8f77c1d2) | [after](https://github.com/elastic/kibana/commit/e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +4.3KB |
| `ml` | 5.8MB | 5.8MB | -15.0B |
| total |  |  | +4.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f466ebf1a3fc33ea09ea51853811fe3a8f77c1d2) | [after](https://github.com/elastic/kibana/commit/e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc) | diff |
| --- | --- | --- | --- |
| `ml` | 66.8KB | 66.8KB | -23.0B |

### History
* :broken_heart: [Build #129367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/) failed fbbf1e923ffedd3532419450f299f581b97d31a9
* :broken_heart: [Build #129297](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/) failed 5e4c520b371288d46408e5909fc1eb9a647e3348
* :broken_heart: [Build #129060](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/) failed ca85c18f2bff94bc69f59b63cb2a42049e3d5e55
* :broken_heart: [Build #129003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/) failed 76b7dffd497b14c40e603af1931c03370beb12ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"}]}
PIPELINE-->

qn895 commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/)
* Commit: f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/85fa7c879110e5861d9704e4f9c4bc3e69821aad) | [after](https://github.com/elastic/kibana/commit/f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/85fa7c879110e5861d9704e4f9c4bc3e69821aad) | [after](https://github.com/elastic/kibana/commit/f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +4.3KB |
| `ml` | 5.9MB | 5.9MB | -15.0B |
| total |  |  | +4.3KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/85fa7c879110e5861d9704e4f9c4bc3e69821aad) | [after](https://github.com/elastic/kibana/commit/f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9) | diff |
| --- | --- | --- | --- |
| `ml` | 66.1KB | 66.1KB | -23.0B |

### History
* :green_heart: [Build #129979](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/) succeeded e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc
* :broken_heart: [Build #129367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/) failed fbbf1e923ffedd3532419450f299f581b97d31a9
* :broken_heart: [Build #129297](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/) failed 5e4c520b371288d46408e5909fc1eb9a647e3348
* :broken_heart: [Build #129060](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/) failed ca85c18f2bff94bc69f59b63cb2a42049e3d5e55
* :broken_heart: [Build #129003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/) failed 76b7dffd497b14c40e603af1931c03370beb12ee

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129003/","number":"129003","commit":"76b7dffd497b14c40e603af1931c03370beb12ee"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"}]}
PIPELINE-->

peteharverson commented:
Don't expect this is related to this PR, but I notice that we don't show a row for any field for which a custom label has been set. For example, after applying the following edit, the field no longer appears in the grid:

![image](https://user-images.githubusercontent.com/7405507/121906400-0072b900-cd23-11eb-8977-66756597971a.png)

peteharverson commented on x-pack/plugins/data_visualizer/public/application/common/components/field_data_row/action_menu/actions.ts:
```diff
@@ -40,6 +52,37 @@ export function getActions(
+    });
+  }
+
+  // Allow to edit index pattern field
```
I see in Discover you get the option to delete runtime fields that have been added to the index pattern? Did you look into providing that functionality? Not a requirement for this PR, but just wondered how much work that would involve?

peteharverson replied:
Note as far I'm aware, the delete action should only be added for runtime fields defined in the index pattern.

qn895 replied:
Added the [delete index pattern field option](https://github.com/elastic/kibana/pull/101316/commits/1afeb8a23dfc135738efb4cb5d43f56a650923fc) here if it's a runtime field and if user has edit index pattern permission. Note that the current EUI behavior automatically makes it a dropdown menu if there's > 2 actions. [This seems to be a strict EUI guideline](https://elastic.github.io/eui/#/tabular-content/tables#adding-actions-to-table) so I think we should revisit where would be the best place for the `delete` button
<img width="1762" alt="Screen Shot 2021-06-14 at 20 24 07" src="https://user-images.githubusercontent.com/43350163/121979233-6bd28000-cd4f-11eb-883d-4ecf26c9ad9e.png">

peteharverson replied:
I think this looks ok, and follows the EUI guidelines with the ellipses icon used when there are more than 2 actions.

qn895 commented:
> Don't expect this is related to this PR, but I notice that we don't show a row for any field for which a custom label has been set. For example, after applying the following edit, the field no longer appears in the grid:

I have updated the PR to fix this issue [here](https://github.com/elastic/kibana/pull/101316/commits/a2eed85c7c7844d3761f63b65db31ddb4bac36fb). 

![2021-06-14 at 20 11](https://user-images.githubusercontent.com/43350163/121979344-97ee0100-cd4f-11eb-8892-1d8e668f04e6.gif)


Also added the index pattern deletion flyout although I don't think the current UI is very ideal
![2021-06-14 at 20 27](https://user-images.githubusercontent.com/43350163/121979387-b3f1a280-cd4f-11eb-8684-21066e657661.gif)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/)
* Commit: c8004f2148ab1394265edfc1ff7423db0663e0d6
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/c8004f2148ab1394265edfc1ff7423db0663e0d6)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/35f962526597ca7b4592b5d30b3566d770f06f41) | [after](https://github.com/elastic/kibana/commit/c8004f2148ab1394265edfc1ff7423db0663e0d6) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/35f962526597ca7b4592b5d30b3566d770f06f41) | [after](https://github.com/elastic/kibana/commit/c8004f2148ab1394265edfc1ff7423db0663e0d6) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +5.6KB |
| `ml` | 5.9MB | 5.9MB | -15.0B |
| total |  |  | +5.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/35f962526597ca7b4592b5d30b3566d770f06f41) | [after](https://github.com/elastic/kibana/commit/c8004f2148ab1394265edfc1ff7423db0663e0d6) | diff |
| --- | --- | --- | --- |
| `ml` | 65.8KB | 65.8KB | -23.0B |

### History
* :green_heart: [Build #130531](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/) succeeded f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9
* :green_heart: [Build #129979](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/) succeeded e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc
* :broken_heart: [Build #129367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/) failed fbbf1e923ffedd3532419450f299f581b97d31a9
* :broken_heart: [Build #129297](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/) failed 5e4c520b371288d46408e5909fc1eb9a647e3348
* :broken_heart: [Build #129060](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/) failed ca85c18f2bff94bc69f59b63cb2a42049e3d5e55

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129060/","number":"129060","commit":"ca85c18f2bff94bc69f59b63cb2a42049e3d5e55"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/)
* Commit: 3a5950f5632b38667b769a5c25ed6c71175287a7
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/3a5950f5632b38667b769a5c25ed6c71175287a7)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/execution/node/784/log)
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/execution/node/830/log)

### History
* :green_heart: [Build #131183](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/) succeeded c8004f2148ab1394265edfc1ff7423db0663e0d6
* :green_heart: [Build #130531](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/) succeeded f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9
* :green_heart: [Build #129979](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/) succeeded e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc
* :broken_heart: [Build #129367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/) failed fbbf1e923ffedd3532419450f299f581b97d31a9
* :broken_heart: [Build #129297](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/) failed 5e4c520b371288d46408e5909fc1eb9a647e3348

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129297/","number":"129297","commit":"5e4c520b371288d46408e5909fc1eb9a647e3348"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/)
* Commit: 4c74b9b6dd90610f0a5b1979abf03994ad49f080
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/4c74b9b6dd90610f0a5b1979abf03994ad49f080)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/execution/node/686/log)
* [Execute xpack-kibana-ciGroup13](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/execution/node/785/log)

### History
* :broken_heart: [Build #131538](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/) failed 3a5950f5632b38667b769a5c25ed6c71175287a7
* :green_heart: [Build #131183](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/) succeeded c8004f2148ab1394265edfc1ff7423db0663e0d6
* :green_heart: [Build #130531](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/) succeeded f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9
* :green_heart: [Build #129979](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/) succeeded e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc
* :broken_heart: [Build #129367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/) failed fbbf1e923ffedd3532419450f299f581b97d31a9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129367/","number":"129367","commit":"fbbf1e923ffedd3532419450f299f581b97d31a9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/","number":"131674","commit":"4c74b9b6dd90610f0a5b1979abf03994ad49f080"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/)
* Commit: 40505a3ba70a08d133635186052d5c7932e12df7
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/40505a3ba70a08d133635186052d5c7932e12df7)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/a816dae16249b1c86a0e1f719d63d3f047192c79) | [after](https://github.com/elastic/kibana/commit/40505a3ba70a08d133635186052d5c7932e12df7) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a816dae16249b1c86a0e1f719d63d3f047192c79) | [after](https://github.com/elastic/kibana/commit/40505a3ba70a08d133635186052d5c7932e12df7) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +6.4KB |
| `ml` | 5.9MB | 5.9MB | -15.0B |
| total |  |  | +6.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/a816dae16249b1c86a0e1f719d63d3f047192c79) | [after](https://github.com/elastic/kibana/commit/40505a3ba70a08d133635186052d5c7932e12df7) | diff |
| --- | --- | --- | --- |
| `ml` | 65.8KB | 65.8KB | -23.0B |

### History
* :broken_heart: [Build #131674](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/) failed 4c74b9b6dd90610f0a5b1979abf03994ad49f080
* :broken_heart: [Build #131538](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/) failed 3a5950f5632b38667b769a5c25ed6c71175287a7
* :green_heart: [Build #131183](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/) succeeded c8004f2148ab1394265edfc1ff7423db0663e0d6
* :green_heart: [Build #130531](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/) succeeded f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9
* :green_heart: [Build #129979](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/) succeeded e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/129979/","number":"129979","commit":"e5385bde34f3fe0d6bf92731d16bbf9569a3cdcc"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/","number":"131674","commit":"4c74b9b6dd90610f0a5b1979abf03994ad49f080"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/","number":"131909","commit":"40505a3ba70a08d133635186052d5c7932e12df7"}]}
PIPELINE-->

peteharverson reviewed (approved):
Tested latest changes and LGTM

jgowdyelastic reviewed (approved):
Added a couple of nit pick comments but generally LGTM

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+      });
+    }
+
+    public async assertActionMenuExists(fieldName: string) {
+      const actionButton = this.rowSelector(fieldName, 'euiCollapsedItemActionsButton');
+      await testSubjects.existOrFail(actionButton);
+    }
```
I find the method name and the actual check a bit confusing: the method name makes me think it's looking for the (opened) action menu, but actually, it asserts that the action menu button exists.
Also, it seems this method is never used, so maybe remove it completely?

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
-    public async assertViewInLensActionEnabled(fieldName: string) {
+    public async ensureActionsMenuOpen(fieldName: string) {
+      await retry.tryForTime(30 * 1000, async () => {
+        const panel = await find.existsByCssSelector('euiContextMenuPanel');
```
Since we don't have much details to check the actions menu popover, we can't be sure at this point if the `euiContextMenuPanel` is an opened action menu (or some other menu popover) nor do we know if it's the popover for the correct table row. So I'd suggest to make sure that all popovers are closed first. Maybe with something like this?
```
public async ensureAllMenuPopoversClosed() {
    await retry.tryForTime(5000, async () => {
        await browser.pressKeys(browser.keys.ESCAPE);
        const popoverExists = await find.existsByCssSelector('euiContextMenuPanel');
        expect(popoverExists).to.eql(false, 'All popovers should be closed');
    });
}
```

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+        await this.ensureActionsMenuOpen(fieldName);
+
+        const button = await find.byCssSelector(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
```
Could use the local `testSubj` variable here.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+          '[data-test-subj="dataVisualizerActionDeleteIndexPatternFieldButton"][class="euiContextMenuItem"]'
+        );
+        await button.click();
+        await this.ensureActionsMenuClosed(fieldName, testSubj);
```
Actually, I think we would like to assert that the action menu is closed.
Currently with `ensureActionsMenuClosed`, we're checking if it's still there and try to close it if needed. This could cause isssue as at this point there's the delete confirm modal open. So I'd suggest to either change this to a simple assertion for the missing/closed action menu or make it instead assert that the delete confirmation model is open.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+      const isEnabled = await testSubjects.isEnabled(actionButton);
+      expect(isEnabled).to.eql(
+        expectedValue,
+        `Expected "Edit index pattern" button for '${fieldName}' to be '${
```
The message doesn't match the assertion in this method:
```suggestion
        `Expected "Explore in lens" button for '${fieldName}' to be '${
```

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos commented on x-pack/test/functional/services/ml/data_visualizer_table.ts:
```diff
@@ -133,17 +153,119 @@ export function MachineLearningDataVisualizerTableProvider(
+    public async clickEditIndexPatternFieldButton(fieldName: string) {
+      await retry.tryForTime(5000, async () => {
+        await this.assertEditIndexPatternFieldButtonEnabled(fieldName, true);
+        await testSubjects.click(
+          this.rowSelector(fieldName, 'dataVisualizerActionEditIndexPatternFieldButton')
+        );
```
We should assert that the click actually did something, e.g. check that the edit field flyout is open.

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/101316/commits/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e

pheyos reviewed (commented):
Great to see functional tests coming as part of this PR :clap: 
Just left a few comments / suggestions.

pheyos commented:
Checking test stability in a [flaky test runner job](https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/1661/) ... no failure in 50 runs :heavy_check_mark:

mattkime reviewed (approved):
lgtm, great to see this being implemented!

qn895 commented on x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_view.tsx:
```diff
@@ -123,9 +124,8 @@ export interface IndexDataVisualizerViewProps {
-    services: { lens: lensPlugin, docLinks, notifications, uiSettings },
-  } = useDataVisualizerKibana();
+  const { services } = useDataVisualizerKibana();
+  const { docLinks, notifications, uiSettings } = services;
```
I think I unpacked the services here because it's being used later in `getActions`

qn895 commented on x-pack/plugins/data_visualizer/public/application/common/components/field_data_row/action_menu/actions.ts:
```diff
@@ -40,6 +52,66 @@ export function getActions(
+        actionFlyoutRef.current = indexPatternFieldEditor?.openEditor({
+          ctx: { indexPattern },
+          fieldName: item.fieldName,
+          onSave: () => {
```
Updated here https://github.com/elastic/kibana/pull/101316/commits/dcd6cfaa9c0c39eb882cbcb147c3bc68e695de51

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/)
* Commit: 6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dec77cfafb0cd557eaf7a6d2ab72280b9801fb6d) | [after](https://github.com/elastic/kibana/commit/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dec77cfafb0cd557eaf7a6d2ab72280b9801fb6d) | [after](https://github.com/elastic/kibana/commit/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +6.3KB |
| `indexPatternFieldEditor` | 22.1KB | 22.2KB | +47.0B |
| `ml` | 5.9MB | 5.9MB | -15.0B |
| total |  |  | +6.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dec77cfafb0cd557eaf7a6d2ab72280b9801fb6d) | [after](https://github.com/elastic/kibana/commit/6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e) | diff |
| --- | --- | --- | --- |
| `ml` | 65.8KB | 65.8KB | -23.0B |

### History
* :green_heart: [Build #131909](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/) succeeded 40505a3ba70a08d133635186052d5c7932e12df7
* :broken_heart: [Build #131674](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/) failed 4c74b9b6dd90610f0a5b1979abf03994ad49f080
* :broken_heart: [Build #131538](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/) failed 3a5950f5632b38667b769a5c25ed6c71175287a7
* :green_heart: [Build #131183](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/) succeeded c8004f2148ab1394265edfc1ff7423db0663e0d6
* :green_heart: [Build #130531](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/) succeeded f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130531/","number":"130531","commit":"f5e9c37b952ff2e1b63f7c50795f8dee4e2e44d9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/","number":"131674","commit":"4c74b9b6dd90610f0a5b1979abf03994ad49f080"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/","number":"131909","commit":"40505a3ba70a08d133635186052d5c7932e12df7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/","number":"133368","commit":"6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e"}]}
PIPELINE-->

pheyos reviewed (approved):
Functional tests LGTM :tada:

qn895 commented:
@elasticmachine merge upstream

qn895 commented:
Checking test stability in a [flaky test runner job](https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/1679/)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133657/)
* Commit: 7279583f6d3026fdc7a6dd35cb3a17a8843f848c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/7279583f6d3026fdc7a6dd35cb3a17a8843f848c)
* [Documentation Changes](https://kibana_101316.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/28162810ad2aa90395c83e483f4cd4dc476b8c11) | [after](https://github.com/elastic/kibana/commit/7279583f6d3026fdc7a6dd35cb3a17a8843f848c) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 236 | 238 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/28162810ad2aa90395c83e483f4cd4dc476b8c11) | [after](https://github.com/elastic/kibana/commit/7279583f6d3026fdc7a6dd35cb3a17a8843f848c) | diff |
| --- | --- | --- | --- |
| `dataVisualizer` | 1.1MB | 1.1MB | +6.3KB |
| `indexPatternFieldEditor` | 22.1KB | 22.2KB | +47.0B |
| `ml` | 5.9MB | 5.9MB | -15.0B |
| total |  |  | +6.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/28162810ad2aa90395c83e483f4cd4dc476b8c11) | [after](https://github.com/elastic/kibana/commit/7279583f6d3026fdc7a6dd35cb3a17a8843f848c) | diff |
| --- | --- | --- | --- |
| `ml` | 65.8KB | 65.8KB | -23.0B |

### History
* :green_heart: [Build #133368](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/) succeeded 6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e
* :green_heart: [Build #131909](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/) succeeded 40505a3ba70a08d133635186052d5c7932e12df7
* :broken_heart: [Build #131674](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/) failed 4c74b9b6dd90610f0a5b1979abf03994ad49f080
* :broken_heart: [Build #131538](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/) failed 3a5950f5632b38667b769a5c25ed6c71175287a7
* :green_heart: [Build #131183](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/) succeeded c8004f2148ab1394265edfc1ff7423db0663e0d6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131183/","number":"131183","commit":"c8004f2148ab1394265edfc1ff7423db0663e0d6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/","number":"131674","commit":"4c74b9b6dd90610f0a5b1979abf03994ad49f080"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/","number":"131909","commit":"40505a3ba70a08d133635186052d5c7932e12df7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/","number":"133368","commit":"6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133657/","number":"133657","commit":"7279583f6d3026fdc7a6dd35cb3a17a8843f848c"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/103156)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/103156">](https://github.com/elastic/kibana/pull/103156) |

This backport PR will be merged automatically after passing CI.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136630/)
* Commit: 7279583f6d3026fdc7a6dd35cb3a17a8843f848c
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101316/7279583f6d3026fdc7a6dd35cb3a17a8843f848c)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136630/execution/node/683/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136630/execution/node/860/log)

### History
* :green_heart: [Build #133657](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133657/) succeeded 7279583f6d3026fdc7a6dd35cb3a17a8843f848c
* :green_heart: [Build #133368](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/) succeeded 6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e
* :green_heart: [Build #131909](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/) succeeded 40505a3ba70a08d133635186052d5c7932e12df7
* :broken_heart: [Build #131674](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/) failed 4c74b9b6dd90610f0a5b1979abf03994ad49f080
* :broken_heart: [Build #131538](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/) failed 3a5950f5632b38667b769a5c25ed6c71175287a7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131538/","number":"131538","commit":"3a5950f5632b38667b769a5c25ed6c71175287a7"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131674/","number":"131674","commit":"4c74b9b6dd90610f0a5b1979abf03994ad49f080"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131909/","number":"131909","commit":"40505a3ba70a08d133635186052d5c7932e12df7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133368/","number":"133368","commit":"6c51e92cd5f9ce63fd4b5bcefb38a46e0b7f633e"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/133657/","number":"133657","commit":"7279583f6d3026fdc7a6dd35cb3a17a8843f848c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/136630/","number":"136630","commit":"7279583f6d3026fdc7a6dd35cb3a17a8843f848c"}]}
PIPELINE-->
